src/many-abci/src/module.rs

```diff
@@ -13,32 +12,44 @@ use many_types::blockchain::{
 };
 use many_types::{blockchain::RangeBlockQuery, SortOrder, Timestamp};
 use once_cell::sync::Lazy;
+use sha2::Digest;
 use std::ops::{Bound, RangeBounds};
 use tendermint::Time;
-use tendermint_rpc::{query::Query, Client};
+use tendermint_rpc::{query::Query, Client, Order};
 
 const MAXIMUM_BLOCK_COUNT: u64 = 100;
 static DEFAULT_BLOCK_LIST_QUERY: Lazy<Query> = Lazy::new(|| Query::gte("block.height", 0));
 
-fn _many_block_from_tendermint_block(block: tendermint::Block) -> Block {
+fn _many_block_from_tendermint_block<C: Client + Sync>(
+    block: tendermint::Block,
+    args: blockchain::ListArgs,
+    client: &C,
+) -> Result<Block, ManyError> {
+    let (count, order, query) = transform_list_args(args)?;
+    let transaction_results_by_id = tx_results(client, count, order, query)?;
     let height = block.header.height.value();
-    let txs_count = block.data.len() as u64;
+    let txs_count: u64 = block
+        .data
+        .len()
+        .try_into()
+        .map_err(|_| ManyError::unknown("Unable to cast usize to u64"))?;
     let txs = block
         .data
         .into_iter()
         .map(|b| {
-            use sha2::Digest;
-            let mut hasher = sha2::Sha256::new();
-            hasher.update(&b);
+            let id = TransactionIdentifier { hash: hash_tx(&b) };
             Transaction {
-                id: TransactionIdentifier {
-                    hash: hasher.finalize().to_vec(),
-                },
-                content: Some(b),
+                id: id.clone(),
+                request: Some(b),
+                response: transaction_results_by_id
```

Comment:
This is inefficient as you execute `find` on every iteration of `data`.

Reply:
@fmorency This is true, although it is still more efficient than making a client call for every iteration of data.  I am open to suggestions here.